src/vec.rs

```diff
@@ -5,34 +5,37 @@ use core::{
     ops, ptr, slice,
 };
 
-/// Workaround forbidden specialization of Drop
-pub trait VecDrop {
-    // SAFETY: drop_with_len will be called to call drop in place the first `len` elements of the buffer.
-    // Only the Owned buffer (`[MaybeUninit<T>; N]`) must drop the items
-    // and the view (`[MaybeUninit<T>]`) drops nothing.
-    // `drop_with_len `assumes that the buffer can contain `len` elements.
-    unsafe fn drop_with_len(&mut self, len: usize);
+pub trait VecBuffer {
```

Comment:
This trait should be sealed.

Reply:
It is sealed because `vec.rs` is already a private module. `Vec` and `VecView` are `pub use vec::{Vec, Vecview}`.

I can add one more layer of sealing in case `vec.rs` is made a public module.

Reply:
I'm also not sure how to get the workaround https://github.com/rust-lang/rust/issues/119015 to work with 2 layers sealing, but it's also no necessary anymore on stable 1.77.0, so we might as well remove it.

Reply:
Okay, fine as is then.